### ISSUE TITLE
BBI: Should the covariate estimator be fitted on the training or test set ? 

### ISSUE THREAD

**Issue by jpaillard** [mind-inria/hidimstat#12]
The covariate estimator $\hat{\mathcal{E}}[X^j|X^{-j}]$ is currently fitted and used for predicting on the test set:
https://github.com/mind-inria/hidimstat/blob/7c3af80e501e2c448dd7b1dbf21f1159c17c514b/hidimstat/compute_importance.py#L251-L265

This could lead to overfitting on the test set and underestimating the conditional importance of covariate $j$. 
Another solution suggested by Ahmad would be to cross-fit on the test set. This could on the contrary lead to poorer fit of the covariate estimator compared to methods like LOCO that leverage the entire train set for accounting for correlations. 

Also, I find the naming `importance_estimator` a bit misleading. It reminds me of the method used for importance estimation (LOCO, SHAP, CPI...) but actually corresponds to the estimator used in CPI to predict $X^j$ from $X^{-j}$

**Comment by bthirion** [comment]
Thx. This should be addressed in 2 PRs.
